lib/dataset/test/logical_reduction_test.cpp

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+#include <gtest/gtest.h>
+#include <limits>
+#include <vector>
+
+#include "scipp/dataset/all.h"
+#include "scipp/dataset/any.h"
+#include "scipp/variable/reduction.h"
+
+using namespace scipp;
+using namespace scipp::dataset;
+
+TEST(AllTest, masked_data_array) {
+  const auto var =
+      makeVariable<bool>(Dimensions{{Dim::Y, 3}, {Dim::X, 2}}, units::m,
+                         Values{true, false, true, true, false, false});
+  const auto mask =
+      makeVariable<bool>(Dimensions{Dim::X, 2}, Values{false, true});
+  DataArray a(var);
+  a.masks().set("mask", mask);
+
+  const auto all_x = makeVariable<bool>(Dimensions{Dim::Y, 3}, units::m,
+                                        Values{true, true, false});
+  const auto all_y =
+      makeVariable<bool>(Dimensions{Dim::X, 2}, units::m, Values{false, false});
+  EXPECT_EQ(all(a, Dim::X).data(), all_x);
+  EXPECT_EQ(all(a, Dim::Y).data(), all_y);
+  EXPECT_FALSE(all(a, Dim::X).masks().contains("mask"));
+  EXPECT_TRUE(all(a, Dim::Y).masks().contains("mask"));
```

Comment:
This is a good example where we would now prefer writing clearly named tests, indicating what we are testing, something like `mask_along_reduction_dim_is_dropped` and `mask_along_preserved_dim_is_preserved`.